sklearn/cross_decomposition/pls_.py

```diff
@@ -375,14 +375,14 @@ def transform(self, X, Y=None, copy=True):
         x_scores if Y is not given, (x_scores, y_scores) otherwise.
         """
         check_is_fitted(self, 'x_mean_')
-        X = check_array(X, copy=copy)
+        X = check_array(X, copy=copy, dtype=np.float64)
```

Comment:
Here we would be casting float32 to float64, which can blow up memory. We should strive to avoid that.